File: monitor.cpp

#include "monitor.h"
#include <iostream>
using std::cout;
using std::endl;

SCREEN *
EloGetScreenByIndex(int index)
{
    static SCREEN toRet;
    static MONITOR toRetMon;
    toRet.nScreenIndex = index;
    toRet.uVendorID = index * 2;
    toRet.uProductID = index * 3;
    toRet.uVersionNumber = index * 4;
    toRet.hCalTouchThread = 0;
    toRet.pMonitor = &toRetMon;
    toRet.pCWndBeamHandler = 0;
    toRet.bIrBeams = 0;
	toRetMon.elo_mon_num = index;
	toRetMon.x = index;
	toRetMon.y = index;
	toRetMon.width = index * 4;
	toRetMon.height = index * 4;
	toRetMon.orientation = 1;
	toRetMon.is_primary = true;
	cout << "Index: " << index << endl;
    return &toRet;
}

File: MONITOR.java
import com.sun.jna.*;
import java.util.List;
import java.util.Arrays;

public class MONITOR extends Structure {
    public int elo_mon_num;
    public int x;
    public int y;
    public int width;
    public int height;
    public int orientation;
    public byte is_primary;

    public MONITOR() {
        super();
    }

    @Override
    protected List<? > getFieldOrder() {
        return Arrays.asList("elo_mon_num", "x", "y", "width", "height", "orientation", "is_primary");
    }

    public MONITOR(Pointer peer) {
        super(peer);
    }

    public static class ByReference extends MONITOR implements Structure.ByReference {
    };

    public static class ByValue extends MONITOR implements Structure.ByValue {
    };
}
